src/Front/DesugarLists.hs

```diff
@@ -1,29 +1,55 @@
 -- | Desugar ListExpr nodes into App nodes
 module Front.DesugarLists
   ( desugarLists
-  ) where
+  )
+where
 
-import qualified Common.Compiler               as Compiler
-import           Front.Ast                      ( Definition(..)
-                                                , Expr(..)
-                                                , Program(..)
-                                                , TopDef(..)
-                                                )
+import qualified Common.Compiler as Compiler
+import Front.Ast (Definition(..), Expr(..), Program(..), TopDef(..), Literal(..))
+import Common.Identifiers as I
 
 -- | Desugar ListExpr nodes inside of an AST 'Program'.
 desugarLists :: Program -> Compiler.Pass Program
-desugarLists (Program decls) = return $ Program $ desugarTop <$> decls
- where
-  desugarTop (TopDef d) = TopDef $ desugarDef d
-  desugarTop t          = t
+desugarLists (Program decls) = return $ Program $ desugarTop <$> decls 
+  where
+    desugarTop (TopDef d) = TopDef $ desugarDef d
+    desugarTop t = t
+    --test = Compiler.unexpected $ show $ (appidhelper [(Lit (LitInt 1)), (Lit (LitInt 2))]) 
+    desugarDef (DefFn v bs t e) = DefFn v bs t $ desugarExpr e
+    desugarDef (DefPat b e) = DefPat b $ desugarExpr e
 
-  desugarDef (DefFn v bs t e) = DefFn v bs t $ desugarExpr e
-  desugarDef (DefPat b e    ) = DefPat b $ desugarExpr e
-
--- | Transform a node of type ListExpr into a node of type App 
+-- | Transform a node of type ListExpr into a node of type App
 -- For ex, (ListExpr [1, 2, 3]) turns into
--- App (App (Id "Cons") (Lit (LitInt 1) )) 
---     (App (App (Id "Cons") (Lit (LitInt 2))) 
+-- App (App (Id "Cons") (Lit (LitInt 1) ))
+--     (App (App (Id "Cons") (Lit (LitInt 2)))
 --          (App (App (Id "Cons") (Lit (LitInt 3))) (id "Nil")))
+
+{-
+appidhelper:: A.ListExpr -> A.ListExpr
+appidhelper [] = [(A.Id (I.Identifier "App")) (A.Id (I.Identifier "Cons")) (A.Id (I.Identifier "Nil"))]
+appidhelper (h:t) = A.Id (I.Identifier "App") (A.Id (I.Identifier "Cons")) h : appidhelper t
+
+ListExpr is not a type, it's a data constructor.
+Data constructors return an instance of a type. 
+ListExpr is a data constructor that returns an instance of type Expr.
+It's not possible to have a function that takes in
+something of type ListExpr and returns something of type ListExpr, because a ListExpr is not a type.
+It is possible to have a function that takes in 
+something of type Expr and returns something of type Expr!
+Then you can special case on instance of ListExpr.
+I think this is what you wanted to do.
+-}
+
+appidhelper:: [Expr] -> Expr
+appidhelper ([]) = (Id (I.Identifier "Nil"))
+appidhelper ((h:t)) = Apply (Apply (Id (I.Identifier "Cons")) h) (appidhelper t)
```

Comment:
One last thing -- do you see how in these parts you are creating an identifier from a raw string literal every time you talk about the Cons or Nil data constructors? To make the code cleaner, define two special identifiers for the two List data constructors inside `src/Common/Identifiers.hs`. 
Then you can use the identifier name instead of writing `(I.Identifer "Cons")` everywhere.
[I had Wei do this for tuples](https://github.com/ssm-lang/sslang/blob/main/src/Common/Identifiers.hs#L283). 
Instead of writing `(Identifier "Pair")` everywhere, he was able to replace those chunks of code with just `tempTuple`.
Please do the same thing for Cons and Nil data constructors. 
**Follow up if anything I said here is confusing!**